docs/devel/offline-virtual-machine.md

```diff
@@ -0,0 +1,234 @@
+# Offline Virtual Machine developer documentation
+
+This document introduces the offline virtual machine object and provides a
+guide how to use it and build upon it.
+
+## What is Offline Virtual Machine
+
+Almost all virtual machine (VM) management systems allow you to manage both running
+and stopped virtual machines. Such system allows you to edit configuration of
+both types of VMs and show its statuses.
+
+To allow building such VM management systems on top of the KubeVirt, the
+OfflineVirtualMachine object is introduced to provide the access to the stopped
+virtual machines. When working with running virtual machines, please see
+the [Virtual Machine object documentation](). The Virtual Machine object is
+designed to work in tandem with the OfflineVirtualMachine and provides the
+configuration and status for running virtual machines.
+
+OfflineVirtualMachine is a Kubernetes [CRD](https://kubernetes.io/docs/concepts/api-extension/custom-resources/), which
+allows for using the Kubernetes machanisms for storing the objects in the etcd,
+exposing REST API and CLI.
+
+## What it does and how to use it
+
+The OfflineVirtualMachine provides the functionality to:
+
+* Store OfflineVirtualMachine in the etcd,
+* Manipulate the OfflineVirtualMachine through the kubectl,
+* Manipulate the OfflineVirtualMachine through the Kubernetes REST API,
+* Watch for changes in the OfflineVirtualMachine and react to them:
+  * Convert the OfflineVirtualMachine to VirtualMachine and thus launch it
+  * Stop VirtualMachine and update status of OfflineVirtualMachine accordingly
+
+### Kubectl interface
+
+The kubectl allows you to manipulate the Kubernetes objects imperatively.
+You can create, delete, update and query objects in the etcd. More details on
+how to use kubectl and what it can do are in the [Kubernetes documentation](https://kubernetes.io/docs/reference/kubectl/overview/).
+
+Following are the examples of working with OfflineVirtualMachine and kubectl:
+
+```bash
+# Define an OfflineVirtualMachine:
+kubectl create -f myofflinevm.yaml
+
+# Start an OfflineVirtualMachine:
+kubectl patch offlinevirtualmachine myvm -p \
+    '{"spec":{"running" :"true"}}'
+
+# Look at OfflineVirtualMachine status and associated events:
+kubectl describe offlinevirtualmachine myvm
+
+# Look at the now created VirtualMachine status and associated events:
+kubectl describe virtualmachine myvm
+
+# Stop an OfflineVirtualMachine:
+kubectl patch offlinevirtualmachine myvm -p \
+    '{"spec":{"running":"false"}}'
+
+# Implicit cascade delete (first deletes the vm and then the ovm)
+kubectl delete offlinevirtualmachine myvm
+
+# Explicit cascade delete (first deletes the vm and then the ovm)
+kubectl delete offlinevirtualmachine myvm --cascade=true
+
+# Orphan delete (The running vm is only detached, not deleted)
+# Recreating the ovm would lead to the adoption of the vm
+kubectl delete offlinevirtualmachine myvm --cascade=false
+```
+
+### The REST API
```

Comment:
Hm. as we are "just" using CRDs, I wonder if this section is needed, after all the CRD API Server dictates how the rest urls look, iow we don't diverge in any way, thus maybe it's not needed.

Thoguhts?

Reply:
I think it is always nice to have the information in one place when learning something. Thus I am voting to leave it here.